src/main/java/SearchEngine.java

```diff
@@ -0,0 +1,109 @@
+import impl.AllStrategyImpl;
+import impl.AnyStrategyImpl;
+import impl.NoneStrategyImpl;
+import strategies.SearchStrategy;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Scanner;
+import java.util.stream.Stream;
+
+public class SearchEngine {
+    Scanner scanner = new Scanner(System.in);
+    ReadManager readManager;
+
+    public SearchEngine(ReadManager readManager) {
+        this.readManager = readManager;
+    }
+
+    public void run() {
+        ArrayList<String> peopleList = readManager.readFile();
+        while (true) {
+            printMenu();
+            String selection = scanner.nextLine();
+            switch (selection) {
+                case "1" -> find(peopleList, mapWords(peopleList));
+                case "2" -> printList(peopleList);
+                case "3" -> {
+                    scanner.close();
+                    System.out.println("\nBye!");
+                    return;
+                }
+                default -> System.out.println("\nIncorrect option! Try again.");
+            }
+        }
+    }
+
+    private Map<String, ArrayList<Integer>> mapWords(ArrayList<String> lines) {
+        Map<String, ArrayList<Integer>> hashMapWords = new HashMap<>();
+        lines.stream()
+                .map(line -> line.trim().split("\s"))
```

Comment:
Please fix your regex and not only here, see - https://www.baeldung.com/java-regex-s-splus

Reply:
fixed